Libplanet/Crypto/ICryptoBackend.cs

```diff
@@ -7,6 +7,20 @@ namespace Libplanet.Crypto
     /// </summary>
     public interface ICryptoBackend
     {
+        /// <summary>
+        /// Creates a signature from <paramref name="messageHash"/> with the corresponding
+        /// <paramref name="privateKey"/>.
+        /// </summary>
+        /// <param name="messageHash">A 32 bytes message hash digest hashed with SHA256 to sign.
+        /// </param>
+        /// <param name="privateKey"><see cref="PrivateKey"/> to sign
+        /// <paramref name="messageHash"/>.
+        /// </param>
+        /// <returns> Created a signature from <paramref name="messageHash"/> with the corresponding
+        /// <paramref name="privateKey"/>.
+        /// </returns>
+        byte[] Sign(HashDigest<SHA256> messageHash, PrivateKey privateKey);
```

Comment:
Why does this take `HashDigest<SHA256>` instead of any arbitrary `byte[]`?

Reply:
Just for consistency with `ICryptoBackend.Verify()`. I don't remember the exact reason why `Verify` take `HashDigest<SHA256>`, but the discussion is here https://github.com/planetarium/libplanet/pull/758#discussion_r366175486. Would it be better `byte[]`?

Reply:
Pair enough.  How about generalizing it over `SHA256`, e.g., `HashDigest<T>`?

Reply:
Changed `ICryptoBackend` to `ICryptoBackend<T>`. 1a1e4d9